### PR TITLE
add indexer gateway client

### PR DESCRIPTION
**Pull request by xiam**
This PRs adds the TypeScript client for Indexer Gateway.

### CODE REVIEW

**Comment by xiam** [comment]
This example:

```ts
import {
  SequenceIndexerGateway
} from '0xsequence/indexergw'

const ACCOUNT_ADDRESS = '0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48'

const indexerGateway = new SequenceIndexerGateway(
  'https://dev-indexer.sequence-dev.app',
)

indexerGateway.getNativeTokenBalance({
  accountAddress: ACCOUNT_ADDRESS,
}).then((res:any) => {
  res.balances.forEach((balance:any) => {
    console.log({balance})
  })
}).catch((error:any) => {
  console.log({error})
})
```

Prints:

```
{
  balance: {
    chainID: 42170,
    balance: {
      accountAddress: '0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48',
      balance: '0'
    }
  }
}
{
  balance: {
    chainID: 10,
    balance: {
      accountAddress: '0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48',
      balance: '371086549230872066'
    }
  }
}
{
  balance: {
    chainID: 43113,
    balance: {
      accountAddress: '0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48',
      balance: '100100000000000000000'
    }
  }
}
{
  balance: {
    chainID: 81457,
    balance: {
      accountAddress: '0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48',
      balance: '0'
    }
  }
}
{
  balance: {
    chainID: 660279,
    balance: {
      accountAddress: '0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48',
      balance: '0'
    }
  }
}
{
  balance: {
    chainID: 11155420,
    balance: {
      accountAddress: '0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48',
      balance: '615667615048000000'
    }
  }
}
{
  balance: {
    chainID: 13473,
    balance: {
      accountAddress: '0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48',
      balance: '8100000000000000000'
    }
  }
}
{
  balance: {
    chainID: 97,
    balance: {
      accountAddress: '0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48',
      balance: '16003756289696138882'
    }
  }
}
{
  balance: {
    chainID: 11155111,
    balance: {
      accountAddress: '0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48',
      balance: '11164700846227378526'
    }
  }
}
{
  balance: {
    chainID: 1101,
    balance: {
      accountAddress: '0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48',
      balance: '6827072461408461'
    }
  }
}
...
```

**Comment by pkieltyka** [comment]
please lets not introduce another "indexergw" package. I know we are doing this on the Go side, but I don't want to do this on the Typescript side.

Please instead keep everything in @0xsequence/indexer and just have packages/indexer/src/indexer.gen.ts and packages/indexer/src/gateway.gen.ts and then for index.ts we can export separately or even we can use the "exports" so developers can `import { IndexerGatewayClient } from '@0xsequence/indexer/gateway'` as an example. But also could also do: `import { IndexerGatewayClient } from '@0xsequence/indexer'` ... we could `export as ....` from index, and bunch of other options.. but lets not introduce "indexergw" package here.

**Review by pkieltyka — DISMISSED** [review]
see comment

**Comment by xiam** [comment]
@corbanbrook This is how output from the latest version looks:

```
[
  {
    chainId: 10,
    result: {
      accountAddress: '0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48',
      balance: '371086549230872066'
    }
  },
  {
    chainId: 81457,
    result: {
      accountAddress: '0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48',
      balance: '0'
    }
  },
  {
    chainId: 137,
    result: {
      accountAddress: '0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48',
      balance: '52408389691068799440368'
    }
  },
  ...
```

File: packages/indexer/src/indexergw.gen.ts [hunk 1]
```diff
@@ -0,0 +1,1375 @@
...
+}
+
+export interface TokenBalancesByContractFilter {
+  contractAddresses: Array<string>
+  accountAddresses?: Array<string>
+  contractStatus?: ContractVerificationStatus
+}
+
+export interface GatewayEtherBalance {
+  chainID: number
```
Comment (corbanbrook) [inline packages/indexer/src/indexergw.gen.ts:515]: should be `chainId`

File: packages/indexer/src/indexergw.gen.ts [hunk 2]
```diff
@@ -0,0 +1,1375 @@
...
+  error: string
+  result: EtherBalance
+}
+
+export interface GatewayNativeTokenBalance {
+  chainID: number
```
Comment (corbanbrook) [inline packages/indexer/src/indexergw.gen.ts:521]: should be `chainId`

File: packages/indexer/src/indexergw.gen.ts [hunk 3]
```diff
@@ -0,0 +1,1375 @@
...
+  error: string
+  result: NativeTokenBalance
+}
+
+export interface GatewayTokenBalance {
+  chainID: number
```
Comment (corbanbrook) [inline packages/indexer/src/indexergw.gen.ts:527]: should be `chainId`

File: packages/indexer/src/indexergw.gen.ts [hunk 4]
```diff
@@ -0,0 +1,1375 @@
...
+}
+
+export interface WALWriterRuntimeStatus {
+  healthOK: boolean
+  startTime: string
+  uptime: number
+  ver: string
+  branch: string
+  commitHash: string
+  chainID: number
```
Comment (corbanbrook) [inline packages/indexer/src/indexergw.gen.ts:218]: should be `chainId` ??

File: packages/indexer/src/indexergw.gen.ts [hunk 5]
```diff
@@ -0,0 +1,1375 @@
...
+
+export interface RuntimeStatus {
+  healthOK: boolean
+  indexerEnabled: boolean
+  startTime: string
+  uptime: number
+  ver: string
+  branch: string
+  commitHash: string
+  chainID: number
```
Comment (corbanbrook) [inline packages/indexer/src/indexergw.gen.ts:207]: should be `chainId` ??

**Review by corbanbrook — CHANGES_REQUESTED** [review]

**Comment by xiam** [comment]
@corbanbrook I fixed the spelling for the newly introduced `chainId` properties; I'll tackle the legacy ones in a follow-up PR.

**Review by corbanbrook — DISMISSED** [review]

**Review by corbanbrook — APPROVED** [review]

**Pull request merged**